Katello/katello
# Fixes #25832 - Show custom syspurpose values in web UI

previous behavior: the dropdowns for syspurpose Addons, Usage and Roles wouldn't show any custom value (not derived from the manifest) which was set on the host. (Though it would show when the field is not in edit mode)

new behavior: the custom value will show in the dropdown/multiselect when set.
bonus: the org would be fetched every time the edit mode was invoked for all syspurpose fields - now the org is cached to make things much snappier

to test:

set a custom syspurpose attribute value on a content host:

```
facet = ::Host.find(1).subscription_facet
facet.purpose_role = 'customrole'
facet.purpose_usage = 'customusage'
facet.purpose_addons = ['customaddon']
facet.save!
```

look at the Content Host details view for the host and observe that these custom values aren't in the dropdowns when setting a new value.

check out this PR and observe that the custom values *do* show in the dropdowns or multiselect. also note that the organizations API is called only once with the new code
